components/tools/samlsso-validator/org.wso2.carbon.identity.tools.saml.validator/src/main/java/org/wso2/carbon/identity/tools/saml/validator/util/SAMLValidatorUtil.java

```diff
@@ -143,12 +144,14 @@ public static String getSAMLRequestFromURL(String url) throws UnsupportedEncodin
     public static Map<String, String> getUserClaimValues(String username, String[] requestedClaims, String profile)
             throws IdentityException {
         try {
-            UserStoreManager userStroreManager =
-                    AnonymousSessionUtil.getRealmByUserName(SAMLSSOUtil.getRegistryService(),
-                            SAMLSSOUtil.getRealmService(),
-                            username).getUserStoreManager();
             username = MultitenantUtils.getTenantAwareUsername(username);
-            return userStroreManager.getUserClaimValues(username, requestedClaims, profile);
+            UserRealm userRealm = AnonymousSessionUtil.getRealmByUserName(SAMLSSOUtil.getRegistryService(),
```

Comment:
Here we are using tenantAwareUsername to get user realm. Hence this approach would not work for tenants. Isn't it always returns super tenant realm?